examples/querying_entity_traits.ipynb

```diff
@@ -0,0 +1,710 @@
+    "updated_ref = manager.register(logo_ref, data_to_publish, PublishAccess.kWrite, context)"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "499f2040f821e2cd"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Re-publishing a minimal entity\n",
+    "\n",
+    "The following example dives deeper into the interaction between `entityTraits`, `managementPolicy` and `resolve` for a generic re-publisher. \n",
+    "\n",
+    "We want to re-publish a minimal entity (i.e. only the traits absolutely required for the given entity reference), with two new/updated traits. One of the traits, `BearerTokenTrait`, might not be supported, and the other trait, `LocatableContentTrait`, might have its properties dictated by the manager."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "b0f6750002fc94ee"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# The minimum set of traits required to publish to this entity reference.\n",
+    "minimum_trait_set = manager.entityTraits(logo_ref, EntityTraitAccess.kWrite, context)\n",
+    "\n",
+    "# Whatever the minimum trait set is, we know we want to publish a location and auth token.\n",
+    "desired_trait_set = minimum_trait_set | {BearerTokenTrait.kId, LocatableContentTrait.kId}\n",
+    "\n",
+    "# Get the set of traits that have properties the manager can persist.\n",
+    "policy_for_desired_traits = manager.managementPolicy(\n",
+    "    desired_trait_set, PolicyAccess.kWrite, context)\n",
+    "\n",
+    "# Filter down the desired traits to only those that are supported.\n",
+    "trait_set_to_publish = desired_trait_set & policy_for_desired_traits.traitSet()\n",
+    "\n",
+    "# We want to keep (the minimum amount of) data from the previous version, except for the values we're going to\n",
+    "# provide.\n",
+    "trait_set_to_keep = trait_set_to_publish - {BearerTokenTrait.kId, LocatableContentTrait.kId}\n",
+    "\n",
+    "# Get the properties that we wish to keep from the current version.\n",
+    "data_to_publish = manager.resolve(logo_ref, trait_set_to_keep, ResolveAccess.kRead, context)\n",
+    "\n",
+    "# Get the manager's policy for dictating trait properties, i.e. which traits the manager can `kDerive` for us.\n",
+    "policy_for_derived_traits = manager.managementPolicy(\n",
+    "    trait_set_to_publish, PolicyAccess.kDerive, context)\n",
+    "\n",
+    "# Check if the manager can derive a location for us.\n",
+    "if LocatableContentTrait.kId in policy_for_derived_traits.traitSet():\n",
+    "    # Imbue an empty LocatableContentTrait, so that the manager is aware in `preflight` that we intend to publish\n",
+    "    # this trait. We will ask the manager to fill in the value for us before calling `register`.\n",
+    "    LocatableContentTrait.imbueTo(data_to_publish)\n",
```

Comment:
Wed need to imbue this (and the other traits in `minimum_trait_set`) regardless as they might not be in `data_to_publish` if they didnt have any properties.

Reply:
I suspect this contrived example is better served treating `LocatableContentTrait` as a special case. It helps to have a docstring explaining the difference between the branching on `managementPolicy`, and it emphasizes that we're publishing the minimal trait set only, but with 1 or 2 extra traits bolted on?

Reply:
Sorry got you - yeah that makes sense. I do wonder if we should split out the generic republishing flows into a separate doc or similar - as they're not using anything new, then we can more easily discuss the subtleties of the whole concept of generic publishing.

Reply:
(We should do that after getting the functionality in though if we do)

Reply:
Generic republishing is now a separate notebook (separate commit in this PR, but can be extracted to a separate PR altogether)